src/client/app/components/LineChartComponent.tsx

```diff
@@ -108,9 +107,30 @@ export default function LineChartComponent() {
 	// See https://community.plotly.com/t/replacing-an-empty-graph-with-a-message/31497 for showing text not plot.
 	if (data.length === 0) {
 		return <h1>{`${translate('select.meter.group')}`}	</h1>;
-	} else if (!enoughData) {
+	} else if (!enoughData || !data[0].x) {
 		return <h1>{`${translate('no.data.in.range')}`}</h1>;
 	} else {
+		// Adjust the min and max values for the x axis
+		// This goes through the list of groups/meters that have data to find the earliest and latest dates.
+		console.log(data);
+		let minDate = '';
+		let maxDate = '';
+		for (const trace of data) {
+			if (trace.x && trace.x.length > 0) {
+				const traceMin = trace.x[0] as string;
+				const traceMax = trace.x[trace.x.length - 1] as string;
+				// Update minX if this is the first trace or has an earlier date
+				if (minDate === '' || traceMin < minDate) {
```

Comment:
Can you help me understand why the string comparison is safe so < and > give the correct answer? Often OED uses moment for comparisons.